src/messages/MOSDECSubOpReadReply.h

```diff
@@ -20,7 +20,7 @@
 
 class MOSDECSubOpReadReply : public MOSDFastDispatchOp {
 private:
-  static constexpr int HEAD_VERSION = 2;
+  static constexpr int HEAD_VERSION = 3;
   static constexpr int COMPAT_VERSION = 1;
```

Comment:
Good point, but we are ok:

* The decode function can decode all 3 versions.
* The encode function can encode version 2 (compatible with version 1 as it is just appending a field) or version 3. It uses a feature bit to determine when the receiving OSD can cope with version 3

I am however being lax about setting the compat_verison - this is always being set to 1. When sending a version 3 message compat_version should be set to 3 because this cannot be decoded by OSDs running version 1 or 2. We should never send them a version 3 message, but it would be better to set compat_version so they assert if this happened. I'll fix that